dockerfiles/scripts/checkout_submodules.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash 
+
+echo "checking out submodule versions"
+
+CUR_PWD=$(pwd)
+cd onnxruntime/cmake/external
+
+# Point to correct onnx-tensorrt
+echo "$1"
+if [ ! "$1" = "8.2" ]
+then
+    cd onnx-tensorrt
+    git remote update
+    if [ "$1" = "8.0" ]
+    then 
+        git checkout "$1"'-GA'
+    fi
+    if [ "$1" = "7.2" ]
+    then 
+        git checkout "$1"'.1'
+    fi 
+    cd ..
+fi
+
+# Point to correct protobuf
+PROTO_VER=$(apt list | grep -oP -m 1 'protobuf-compiler.* \K3[\.\d]+')
```

Comment:
`protoc --version throws protoc command not found. `
This is because protobuf-compiler is installed when we do apt-get update. If I run
`apt-get install protobuf-compiler` then protoc is found, but the script currently doesn't do that. 
So, I'm grepping from the apt list the protobuf-compiler version number 3.x.x. 

What is preferred?

Reply:
For what it's worth, using `apt` in shell scripts/expressions is typically not recommended. The following warning is printed in bash when running the above line: `WARNING: apt does not have a stable CLI interface. Use with caution in scripts.`

I believe `apt-get`, `apt-cache`, and `dpkg` all produce stable output that can be parsed. So maybe `dpkg -l protobuf-compiler | grep ...` may work.

However, querying an installed protoc just seems better.

Reply:
Thanks Adrian! I'll try that out.